src/tools/perf/ucp_tests.cc

```diff
@@ -108,6 +108,27 @@ class ucp_perf_test_runner {
         return UCS_OK;
     }
 
+    void send_tag_complete()
+    {
+        m_outstanding--;
+    }
+
+    static void send_tag_cb(void *request, ucs_status_t status)
+    {
+        ucp_perf_request_t *r = (ucp_perf_request_t*)request;
+        ucp_perf_test_runner *sender = (ucp_perf_test_runner*)r->context;
```

Comment:
maybe encapsulate type cast inside `ucp_perf_request_t`? I would be like
`ucp_perf_request_t::convert_ptr(request).runner().send_tag_complete();`

Reply:
not sure it's worth it, it's single line anyway.. i can change it to use reinterpret_cast<>, though.

Reply:
Up to you. it could be just better readable and object oriented...